src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt

```diff
@@ -5,87 +5,96 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.Pris
 import java.time.Clock
 import java.time.LocalDate
 
+typealias EligibilityCheck = (PrisonerSearchPrisoner) -> Boolean
+operator fun EligibilityCheck.not(): EligibilityCheck = { !this(it) }
+infix fun  EligibilityCheck.describedAs(message: String): Pair<EligibilityCheck, String> = this to message
+
 @Service
 class EligibilityService(
   private val clock: Clock,
 ) {
 
+  val checks = listOf(
+    !isPersonParoleEligible() describedAs "is eligible for parole",
+    hasCorrectLegalStatus() describedAs "has incorrect legal status",
+    !isOnIndeterminateSentence() describedAs "is on indeterminate sentence",
+    hasConditionalReleaseDate() describedAs "has no conditional release date",
+    isEligibleIfOnAnExtendedDeterminateSentence() describedAs "is on non-eligible EDS",
+    hasActivePrisonStatus() describedAs "is not active in prison",
+    hasReleaseDateInTheFuture() describedAs "release date in past",
+    !isRecallCase() describedAs "is a recall case",
+  )
+
   fun isEligibleForCvl(prisoner: PrisonerSearchPrisoner): Boolean {
-    return !prisoner.isPersonParoleEligible() &&
-      prisoner.hasCorrectLegalStatus() &&
-      !prisoner.isOnIndeterminateSentence() &&
-      prisoner.hasConditionalReleaseDate() &&
-      prisoner.isEligibleIfOnAnExtendedDeterminateSentence() &&
-      prisoner.hasActivePrisonStatus() &&
-      prisoner.hasEligibleReleaseDate() &&
-      !prisoner.isRecallCase()
+    return getIneligiblityReasons(prisoner).isEmpty()
+  }
+
+  fun getIneligiblityReasons(prisoner: PrisonerSearchPrisoner): List<String> {
+    return checks.mapNotNull { (test, message) -> if (!test(prisoner)) message else null }
   }
 
-  private fun PrisonerSearchPrisoner.isPersonParoleEligible(): Boolean {
-    if (this.paroleEligibilityDate != null) {
-      if (this.paroleEligibilityDate.isAfter(LocalDate.now(clock))) {
-        return true
+
+  private fun isPersonParoleEligible(): EligibilityCheck = early@{
+    if (it.paroleEligibilityDate != null) {
+      if (it.paroleEligibilityDate.isAfter(LocalDate.now(clock))) {
+        return@early true
       }
     }
-    return false
+    return@early false
```

Comment:
These labels are needed to return from a lambda expression - could tidy these up using `when`s